### ISSUE TITLE
SendTextPlus stopped working

### ISSUE THREAD

**Issue by miguelroboso** [randy3k/SendTextPlus#35]
Hello,

I am experiencing a weird issue. Until a while ago ST+ was working (besides the issue I posted in another thread). In the last two weeks, though, it stopped working completely. No matter what program I chose to send text to (terminal, RStudio, RStudio on Safari, Chrome, REPL) ST+ just does not do anything. 

Anyone else experiencing this?

**Comment by randy3k** [comment]
Is there any error messages in the console? What do you mean by not working?

**Comment by miguelroboso** [comment]
Hi Randy,

no, the console does not show any message (Screenshot attached).
By "not working" I mean that I click CMD+Enter, the cursor goes to the next line, but nothing is sent to R.
![screen shot 2016-06-08 at 18 44 04](https://cloud.githubusercontent.com/assets/17713880/15916571/0857cb66-2da9-11e6-9f60-9be86d19578d.png)

**Comment by randy3k** [comment]
would you run `sublime.log_commands(True)` in the console and then check if 

```
command: send_text_plus
```
shows up when you hit cmd+enter.

**Comment by miguelroboso** [comment]
Hi Randy, 

thanks. 

This is what I get when I hit cmd+enter (disregard the drags :) ):

`>>> sublime.log_commands(True)
command: drag_select {"event": {"button": 1, "x": 540.3984375, "y": 310.83203125}}
command: send_text_plus
command: send_text_plus
command: send_text_plus
command: send_text_plus
command: send_text_plus
command: send_text_plus
command: send_text_plus
command: drag_select {"event": {"button": 1, "x": 189.55859375, "y": 1115.70703125}}
command: drag_select {"event": {"button": 1, "x": 621.37890625, "y": 1120.4140625}}
command: drag_select {"event": {"button": 1, "x": 621.37890625, "y": 1120.4140625}}`

**Comment by randy3k** [comment]
it has no problem in triggering `send_test_plus` and there are no error messages. Odd..

**Comment by miguelroboso** [comment]
Maybe I messed up something in the configuration? Is there a way to get a clean default configuration file? I tried to uninstall and re-install but no luck.

**Comment by randy3k** [comment]
This is strange that both applescript and terminal do not work while there are no error messages...
I don't think it is a problem of the configuration file. There is a fundamental

**Comment by miguelroboso** [comment]
Ok, I need to apologize here: I just checked all of them again: Terminal, RStudio, RStudio server in Safari, and actually Terminal works. Everything else still does not, but terminal works.

**Comment by randy3k** [comment]
Did RStudio ever work? You can use a sufficient updated version of RStudio.

**Comment by miguelroboso** [comment]
yes, it worked up to a month ago (remember my post about the focus not coming back to the source)

**Comment by randy3k** [comment]
It is completely out of my mind. The only explanation that is that your RStudio is broken as  terminal works while RStudio doesn't and there are no errors.

**Comment by miguelroboso** [comment]
Ok, I will re-install RStudio locally and see if that solves it.
But how about RStudio Server on a Safari window? That is a machine that has not had RStudio nor R updated in the past year. I must have ruined something in some way...

Thanks for your help!

**Comment by randy3k** [comment]
RStudio Server should work as long as your safari or chrome is working. I cannot think of any reasons that Terminal works while all other don't...

Um...could you check that apple script editor works or not?

Open Applications/Utilities/Script Editor.app

```
tell application "RStudio" to cmd "a <- 1"
```

**Comment by miguelroboso** [comment]
RStudio server does not work. 

Bingo: 

`tell: command not found`

How the hek did I end up without "tell"?

**Comment by randy3k** [comment]
I think you misunderstood me.

Either open Applications/Utilities/Script Editor.app, 
then run 
```
tell application "RStudio" to cmd "a <- 1"
```

or open terminal and run
```
osascript -e 'tell application "RStudio" to cmd "a <- 1"'
```

But not running `tell application "RStudio" to cmd "a <- 1"` on terminal.

**Comment by miguelroboso** [comment]
Yeah, I was figuring out that right now...
However, both ways I get this error:
30:35: syntax error: A “"” can’t go after this identifier. (-2740)'

**Comment by randy3k** [comment]
Oh, this is the reason. The communication between Sublime to RStudio (via applescript) is broken. I have no idea what causes it.

**Comment by miguelroboso** [comment]
I tried to do this: 

https://gist.github.com/wch/3789190

And it works. You are right in that though: the cmd directive seems not to work anymore. That makes me think though, how is it passed to Safari/Chrome? Still through applescript?

**Comment by randy3k** [comment]
Yes, everything of Mac is via applescript.

https://gist.github.com/wch/3789190 actually uses `System Events` but not RStudio itself.

**Comment by miguelroboso** [comment]
Good point. Now I have to understand why is my cmd directive broken for RStudio, and how to fix it.

**Comment by randy3k** [comment]
I just realized that the new version Safari has an "new" security that breaks SendTextPlus. You have to go to "Developer -> Allow JavaScript from Apple Event"

**Comment by miguelroboso** [comment]
Yep, that solved it for RStudio on Safari. So it might be that there were two independent events, one due to applescript not sending the "cmd" directive anymore, and the other due to safari being "safe".

**Comment by randy3k** [comment]
it seems to be the case. Hope you could to figure it out.
